app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -0,0 +1,21 @@
+###
+# Copyright 2016 - 2021 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+class Users::OmniauthCallbacksController < Devise::OmniauthCallbacksController
+  def okta
+    # debugger
+    logger.debug "OmniauthCallbacksController#okta #{request.env['omniauth.auth'].inspect}"
+
+    user = User.from_omniauth request.env['omniauth.auth']
+    sign_in_and_redirect user, event: :authentication # this will throw if @user is not activated
+    set_flash_message(:notice, :success, kind: 'OKTA') if is_navigational_format?
+  end
+
+  def failure
+    logger.warn "OmniauthCallbacksController#failure #{request.env['omniauth.auth']}"
```

Comment:
Should this be `logger.error`?

Reply:
Sure. I made it warning since the user will get a meaningful error and there could be many reasons for the flow other than a server error. If we want we can notify us in early days, i think they are relatively rare

Reply:
sounds fine then, just felt odd to have a method called failure with only a warning in it